src/main/java/com/google/api/generator/gapic/composer/grpc/GrpcServiceStubClassComposer.java

```diff
@@ -335,4 +447,120 @@ private LambdaExpr createRequestParamsExtractorClassInstance(Method method) {
+      Method method,
+      List<Statement> classStatements,
+      VariableExpr paramsVarExpr,
+      List<Expr> bodyExprs,
+      VariableExpr requestVarExpr) {
+    ImmutableList<RoutingHeader> routingHeaders = method.routingHeaders().routingHeadersList();
+    for (int i = 0; i < routingHeaders.size(); i++) {
+      RoutingHeader routingHeader = routingHeaders.get(i);
+      MethodInvocationExpr requestFieldGetterExpr =
+          createRequestFieldGetterExpr(requestVarExpr, routingHeader.field());
+      Expr routingHeaderKeyExpr =
+          ValueExpr.withValue(StringObjectValue.withValue(routingHeader.name()));
+      // TODO: Create proper snake style upper case name
+      String pathTemplateName =
+          String.format("PATH_TEMPLATE_%s_%s", method.name().toUpperCase(), i);
+      TypeNode pathTemplateType =
+          TypeNode.withReference(ConcreteReference.withClazz(PathTemplate.class));
+      Variable pathTemplateVar =
+          Variable.builder().setType(pathTemplateType).setName(pathTemplateName).build();
+      Expr routingHeaderPatternExpr = VariableExpr.withVariable(pathTemplateVar);
+      VariableExpr pathTemplateVarExpr =
+          VariableExpr.builder()
+              .setVariable(pathTemplateVar)
+              .setIsDecl(true)
+              .setIsStatic(true)
+              .setIsFinal(true)
+              .setScope(ScopeNode.PRIVATE)
+              .build();
+      ValueExpr valueExpr =
+          ValueExpr.withValue(StringObjectValue.withValue(routingHeader.pattern()));
+      Expr pathTemplateExpr =
+          AssignmentExpr.builder()
+              .setVariableExpr(pathTemplateVarExpr)
+              .setValueExpr(
+                  MethodInvocationExpr.builder()
+                      .setStaticReferenceType(pathTemplateType)
+                      .setMethodName("create")
+                      .setArguments(valueExpr)
+                      .setReturnType(pathTemplateType)
+                      .build())
+              .build();
+      Statement pathTemplateClassVar = ExprStatement.withExpr(pathTemplateExpr);
+      classStatements.add(pathTemplateClassVar);
+      MethodInvocationExpr addParamsMethodExpr =
+          MethodInvocationExpr.builder()
+              .setStaticReferenceType(
+                  TypeNode.withReference(ConcreteReference.withClazz(RoutingHeaderHelper.class)))
+              .setMethodName("addParams")
+              .setArguments(
+                  paramsVarExpr,
+                  requestFieldGetterExpr,
+                  routingHeaderKeyExpr,
+                  routingHeaderPatternExpr)
+              .build();
+
+      bodyExprs.add(addParamsMethodExpr);
+    }
+  }
+
+  private void addRequestParamsForHttpBindings(
+      Method method,
+      VariableExpr paramsVarExpr,
+      List<Expr> bodyExprs,
+      VariableExpr requestVarExpr) {
+    // TODO: Are these params used for anything else other than implicit dynamic routing?
```

Comment:
Hmm... can we answer this? I think it's possible that they are used for other purposes? What we have been doing is really constructing the request parameters? It's just that we are directly formulating the implicit routing header out of the request params?

Reply:
Answered that the params are not used for any other purposes.